airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/steps/gradle.py

```diff
@@ -170,8 +165,8 @@ async def _run(self, *args: Any, **kwargs: Any) -> StepResult:
         # Mount only the code needed to build the connector.
         gradle_container = (
             gradle_container_base
-            # Copy the local maven repository and force evaluation of `with_whole_git_repo` container.
-            .with_directory(self.LOCAL_MAVEN_REPOSITORY_PATH, await with_whole_git_repo.directory(self.LOCAL_MAVEN_REPOSITORY_PATH))
+            # Copy the gradle home directory and force evaluation of `with_whole_git_repo` container.
+            .with_directory(self.GRADLE_HOME_PATH, await with_whole_git_repo.directory(self.GRADLE_HOME_PATH))
```

Comment:
Just to be clear:
The `.gradle` folder from the host will not be mounted as it's excluded[ when calling `get_repo_dir`](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/models/contexts/pipeline_context.py#L58).
I think your intent is that `self._get_gradle_command("help", *warm_dependency_cache_args)` will seed `.gradle`. Which works.
Can we make sure that `.gradle` is empty before calling `f"(rsync -a --stats --mkpath {self.GRADLE_DEP_CACHE_PATH}/ {self.GRADLE_HOME_PATH} || true)",` , (with a `without_directory` call), it'll guarantee that any refactoring an the `get_repo_dir` method will not alter this behavior and lead to corrupted `.gradle` folder mixing host file and container files.

Reply:
Good idea!